src/test/java/org/beanpod/switchboard/controller/DecoderControllerTest.java

```diff
@@ -130,8 +142,34 @@ final void testDeleteDecoder(){
     //When a decoder is unavailable in the DB
     @Test
     final void testDeleteDecoderNotExisting(){
-        assertThrows(DeviceNotFoundException.class, () -> {
+        assertThrows(ExceptionType.DeviceNotFoundException.class, () -> {
             decoderController.deleteDecoder("Not Available decoder");
         }, "DeviceNotFoundException should have been thrown.");
     }
+
+
+    //When a encoder is available in the DB
+    @Test
+    final void testUpdateDecoder(){
+
+        when(decoderService.findDecoder("1")).thenReturn(java.util.Optional.of(decoder1));
+        decoder1.getInputs().clear();
+        when(decoderService.save(decoder1)).thenReturn(decoder1);
+        DecoderDTO decoderDTO1 = decoderMapper.toDecoderDTO(decoder1);
+        ResponseEntity<DecoderDTO> response = decoderController.updateDecoder("1", decoderDTO1);
+
+        assertEquals(200, response.getStatusCodeValue());
+        assertTrue(response.getBody().getInputs().isEmpty());
+    }
+
+    //Test exceptions when updating encoder
+    @Test
+    final void testUpdateDecoderExceptions(){
+        DecoderDTO decoderDTO1 = decoderMapper.toDecoderDTO(decoder1);
+
+        //When device is unavailable in the DB
+        assertThrows(ExceptionType.DeviceNotFoundException.class, () -> {
+            decoderController.updateDecoder("3", decoderDTO1);
+        }, "DeviceNotFoundException should have been thrown.");
```

Comment:
I think this test might be passing for the wrong reasons. There should be a mock here that explicitly mocks the `decoderService` to return an empty list.

Reply:
fixed